### PR TITLE
✨(calendar) display calendar invites

### LINKED ISSUE #433: Add calendar invite (VCALENDAR) parsing and display
Add support for parsing and displaying calendar invites (VCALENDAR/iCalendar format) in the message viewer with a proper UI component.

   ## Current State

   After the MIME parsing fixes in #432 , calendar invites are now properly classified:
   - `text/calendar` parts are correctly included in message body (not as attachments)
   - `.ics` attachment files are downloadable
   - Google Calendar invites display their HTML content correctly

   However, there's no special handling for calendar invite content. Users see either:
   - Raw VCALENDAR text (if no HTML alternative is provided)
   - Generic HTML (for most invites, like Google or Microsoft)
   - A downloadable .ics file

   ## Proposed Enhancement

   Add a calendar invite UI component that:

   1. **Detects calendar content**
      - Check `textBody` array for parts with `type: 'text/calendar'`
      - Check attachments for `.ics` files

   2. **Parses VCALENDAR data**
      - Extract meeting details: title, date/time, location, description
      - Parse attendees list and their RSVP status
      - Handle timezones correctly
      - Support recurring events

   3. **Display rich UI**
      - Show meeting card with formatted date/time
      - Display location with map link
      - List attendees
      - Show event description
      - Add to calendar button (future calendar integration)

   4. **Optional: RSVP handling**
      -   (future calendar integration)

   ## Implementation Notes

   - Calendar parts are already available in parsed message data at `textBody[].type === 'text/calendar'`
   - .ics files should remain downloadable as attachments

   ## Example VCALENDAR structure

   ```
   BEGIN:VCALENDAR
   VERSION:2.0
   METHOD:REQUEST
   BEGIN:VEVENT
   DTSTART;TZID=Europe/Paris:20251121T113000
   DTEND;TZID=Europe/Paris:20251121T123000
   SUMMARY:Mosacloud
   LOCATION:Paris
   ORGANIZER;CN=xxxx
   ATTENDEE;CN=riel@mosacloud.eu;RSVP=TRUE:mailto:riel@mosacloud.eu
   END:VEVENT
   END:VCALENDAR
   ```

### PR DESCRIPTION
**Pull request by sylvinus**
Fixes #433. We only parse them from the frontend for now, and don't provide options to RSVP but this will come with the upcoming Calendars integration.

In the scope of this PR we could add configurable buttons to have "add to calendar" links to popular providers.

Capture d’écran 2026-01-15 à 01 47 31

## Summary by CodeRabbit

* **New Features**
  * Calendar invite previews inline in message footers with event details, organizer, location, recurrence, status badges, attendee lists (expandable), and duplicate invite deduplication.
  * Download .ics files from message previews; cancellation banner and clearer attendee status indicators.

* **Localization**
  * Expanded English and French translations for calendar UI, statuses, recurrence phrases, and loading/error messages.

* **UI / Style**
  * New styling, loading, empty and error states for calendar invite display.

* **Tests**
  * Comprehensive ICS parsing and formatting test suite covering real-world samples and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+                    <Icon name="event" type={IconType.OUTLINED} />
+                </div>
+                <div className="calendar-invite__title-section">
+                    <h3 className="calendar-invite__title">{event.summary}</h3>
+                    {event.status && (
+                        <span
+                            className={`calendar-invite__event-status calendar-invite__event-status--${event.status.toLowerCase()}`}
+                        >
+                            {t(`event.status.${event.status.toLowerCase()}`)}
+                        </span>
```
Comment (jbpenrath) [inline src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx]: There is a `Badge` component that would fit well to display this information

File: src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/_index.scss
Comment (jbpenrath) [inline src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/_index.scss:1]: IMO we could improve the layout by mainly reducing the vertical spacing between elements to reduce the block height.

**Review by jbpenrath — APPROVED** [review]
Great feature!

File: src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+            };
+        case "NEEDS-ACTION":
+        default:
+            return {
+                icon: "schedule",
+                label: t("Awaiting response"),
+                className: "calendar-invite__attendee-status--pending",
+            };
+    }
+}
```
Comment (jbpenrath) [inline src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx]: I feel all those calendar utils could be move into a CalendarHelper module to lighten this file.

File: src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+    const { t } = useTranslation();
+    return (
+        <Button
+            size="small"
+            variant={variant}
+            icon={<Icon name="download" type={IconType.OUTLINED} />}
+            href={downloadUrl}
+            download={name}
+        >
+            {t("Download .ics file")}
```
Comment (jbpenrath) [inline src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx]: The current string is IMO too technical. 
```suggestion
            {t("Download calendar event")}
```
```suggestion
            {t("Download event")}
```
```suggestion
            {t("Download invitation")}
```

File: src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx
Comment (jbpenrath) [inline src/frontend/src/features/layouts/components/thread-view/components/calendar-invite/index.tsx:1]: Not a mandatory change request, just for information: We could use `useQuery` hook from `react-query` to cache response without all this logic.
